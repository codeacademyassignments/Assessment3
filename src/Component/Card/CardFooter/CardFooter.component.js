import React from 'react';
import './CardFooter.component.css';

const CardFooter = (props) => {
  const heartUrl = props.liked ? 'heart-red.svg' : 'heart-black.svg';
  console.log(props);
  return (
    <div className="cardFooter">
      <div>
        <p className="rating">{props.rating}</p>
      </div>
      <div className="heartImage">
        <img
          src={require(`../../../Icons/${heartUrl}`)}
          alt="heart-black"
          className="heart"
          onClick={() => {
            alert('clicked');
            props.onHeartClick(props.liked);
          }}
        />
      </div>
    </div>
  );
};

CardFooter.defaultProps = {
  liked: false,
};

export default CardFooter;
