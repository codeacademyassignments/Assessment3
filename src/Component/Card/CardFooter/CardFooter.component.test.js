import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CardFooter from './CardFooter.component';


describe('CardFooter', () => {
  it('snapshot should match', () => {
    const tree = renderer.create(<CardFooter liked title="Harry Potter and prisoner of azekaban" />);
    expect(tree).toMatchSnapshot();
  });
});

describe('mock function', () => {
  const onHeartClickMock = jest.fn();
  const wrapper = shallow(<CardFooter liked rating={4.5} onHeartClick={onHeartClickMock} />);
  it('should call onHeartClickMock when heart clicked', () => {
    wrapper.find('.heart').simulate('click');
    expect(onHeartClickMock).toHaveBeenCalledWith(true);
    expect(onHeartClickMock).toHaveBeenCalledTimes(1);
  });
});
