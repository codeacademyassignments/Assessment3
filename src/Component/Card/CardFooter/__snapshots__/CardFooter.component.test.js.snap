// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter snapshot should match 1`] = `
<div
  className="cardFooter"
>
  <div>
    <p
      className="rating"
    />
  </div>
  <div
    className="heartImage"
  >
    <img
      alt="heart-black"
      className="heart"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "heart-red.svg",
        }
      }
    />
  </div>
</div>
`;
